fixing ids

Z3rca opened the pull request:
Issue:
Code is very unreadable when we don't add the description of a perk, text to a widget, dialog options or similar but load them up from the tor_strings. The cross referencing drives one mad

second issue: 
Game Text writers and Game Content Designers, are not able to find the spots easily to change the meaning of the texts.

OCD issue:
we have very many pattern of defining our own ids, which i want to transform to a easily identifiable solution.
all regular texts gets an id, which is starting with tor_
all translation tags get the str_[same id as the text] to have a coherent structure. 

this PR is a first draft for changing ALL text to fit this new structure. I so far converted all Code and all xmls in the tor_strings to the new format. there might be a few blind spots here and there but i find them and clear them up.

changed:
transfered all ids to new tor_...  ids and and all localizations to str_tor...
edit all code instances
tried use TextObject patterns were its applicable. 

experimental : 
Career perk and choice groups follow new experimental pattern: it logs if the code text is different from the tor_strings, so it gets easier to spot mistakes. The ground truth sits in the tor_strings not the code.

Z3rca commented on CSharpSourceCode/Extensions/MissionExtensions.cs:
```diff
@@ -5,6 +5,14 @@ namespace TOR_Core.Extensions
+        {
+            var abilityManagerMissionLogic = mission.GetMissionBehavior<AbilityManagerMissionLogic>();
+            return abilityManagerMissionLogic.CurrentState == AbilityModeState.Casting ||
+                   abilityManagerMissionLogic.CurrentState == AbilityModeState.Casting;
```
why? have to change this lol

Z3rca committed:
add tor_ to all ids

Z3rca committed:
warrior priest test

Z3rca committed:
test choice groups and spanish

Z3rca committed:
wip warrior priest test for cleaner layout

Z3rca committed:
refactor for easier naming and finding issues

Z3rca committed:
externalized choice group texts

Z3rca committed:
fix for all careers display text

Z3rca committed:
Test

Z3rca committed:
enum string variant approach

Z3rca committed:
more enums

Z3rca committed:
switch code to new naming convention

Z3rca committed:
hireling activities

Z3rca committed:
fixes from warnings

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomDialogs/CareerSwitchCampaignBehavior.cs:
```diff
@@ -44,21 +44,21 @@ private void BlackGrailKnightDialogLines(CampaignGameStarter starter)
-            starter.AddDialogLine("blackgrail_receive", "black_grail_player_ready", "black_grail_player_playcutscene", "{=tor_blackgrail_receive_str}The truth is: The Bretonnes, proud humans follow the whisperings of an Elf. The Lady is nothing more than a Elven Witch, not a goddess or saint. Our whole Kingdom is build upon the Lie of Lady. I will clean you from these lies, drink from this grail, ...
-            starter.AddDialogLine("bloodkiss_cutscene", "black_grail_player_playcutscene", "close_window", "{=tor_blackgrail_receive_end_str}Rise my young Knight. Let us clean this world from this non sense.", null, null);
+            starter.AddPlayerLine("bloodkiss_inqury_declined", "black_grail_player_ready", "black_grail_player_notready", "{=str_tor_blackgrail_receive}On a second thought, I need to think about it.", () => inquiryDeclined, null);
+            starter.AddDialogLine("blackgrail_receive", "black_grail_player_ready", "black_grail_player_playcutscene", "{=str_tor_blackgrail_receive}The truth is: The Bretonnes, proud humans follow the whisperings of an Elf. The Lady is nothing more than a Elven Witch, not a goddess or saint. Our whole Kingdom is build upon the Lie of Lady. I will clean you from these lies, drink from this grail, ...
```
Localization id is identical for this dialog line and the player line above.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomDialogs/CareerSwitchCampaignBehavior.cs:
```diff
@@ -83,24 +83,24 @@ void OnBlackGrailReceived()
 
 
-            starter.AddPlayerLine("bloodkiss_inqury_declined2", "bloodkiss_blooddragon_player_ready", "bloodkiss_blooddragon_player_notready", "{=tor_blackgrail_receive_str}On a second thought, I need to think about it.", () => inquiryDeclined, null);
+            starter.AddPlayerLine("bloodkiss_inqury_declined2", "bloodkiss_blooddragon_player_ready", "bloodkiss_blooddragon_player_notready", "{=str_tor_blackgrail_receive}On a second thought, I need to think about it.", () => inquiryDeclined, null);
```
This is is the same string as line 93, it can use the same localization id. (Blackgrail id for bloodkiss is somewhat confusing)

"On second thought...." no 'a' before second.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomDialogs/CareerSwitchCampaignBehavior.cs:
```diff
@@ -129,21 +129,21 @@ bool IsEligibleForBloodKissBloodDragon()
         private void VampireCountDialogLines(CampaignGameStarter starter)
         {
-            starter.AddPlayerLine("request_bloodkiss", "lord_talk_speak_diplomacy_2", "bloodkiss_root", "{=tor_bloodkiss_request_str}My lord von Carstein, I wish to shed my mortal shell and serve you eternally.", IsEligibleForBloodKissCarstein, null);
+            starter.AddPlayerLine("request_bloodkiss", "lord_talk_speak_diplomacy_2", "bloodkiss_root", "{=str_tor_bloodkiss_request}My lord von Carstein, I wish to shed my mortal shell and serve you eternally.", IsEligibleForBloodKissCarstein, null);
```
"My lord von Carstein" feels weird, would he be referred to as "von Carstein" typically?
Would this be better with a variable that's set to the OneToOneConversationHero's name so it reads as "My lord, Mannfred von Carstein, ....."?

If only manny can be kissed, then could also stick his name in directly.

Z3rca replied:
well put those discussions to Piero, i am just code monkey

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomDialogs/CareerSwitchCampaignBehavior.cs:
```diff
@@ -129,21 +129,21 @@ bool IsEligibleForBloodKissBloodDragon()
             starter.AddDialogLine("bloodkiss_player_prompt", "bloodkiss_player_prompt", "bloodkiss_player_ready", "...", null, () => DisplayPrompt(TORCareers.MinorVampire, OnBloodKissRecieved));
 
-            starter.AddPlayerLine("bloodkiss_inqury_declined2", "bloodkiss_player_ready", "bloodkiss_player_notready", "{=tor_blackgrail_receive_str}On a second thought, I need to think about it.", () => inquiryDeclined, null);
+            starter.AddPlayerLine("bloodkiss_inqury_declined2", "bloodkiss_player_ready", "bloodkiss_player_notready", "{=str_tor_blackgrail_receive}On a second thought, I need to think about it.", () => inquiryDeclined, null);
```
All of these bloodkiss_declinedX can use the same string & localization id unless these are intended to have slightly different variants between them.

Z3rca replied:
they are intended to be slightly different. We are not done with this "feature yet :D

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomDialogs/CareerSwitchCampaignBehavior.cs:
```diff
@@ -176,21 +176,21 @@ void OnBloodKissRecieved()
             starter.AddDialogLine("bloodkiss_necrach_player_prompt", "bloodkiss_necrach_player_prompt", "bloodkiss_nechrach_player_ready", "...", null, () => DisplayPrompt(TORCareers.Necrarch, OnNecrachBloodKissRecieved));
 
-            starter.AddPlayerLine("bloodkiss_inqury_declined2", "bloodkiss_player_ready", "bloodkiss_player_notready", "{=tor_blackgrail_receive_str}On a second thought, I need to think about it.", () => inquiryDeclined, null);
+            starter.AddPlayerLine("bloodkiss_inqury_declined2", "bloodkiss_player_ready", "bloodkiss_player_notready", "{=str_tor_blackgrail_receive}On a second thought, I need to think about it.", () => inquiryDeclined, null);
```
blackgrail id again

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomDialogs/CareerSwitchCampaignBehavior.cs:
```diff
@@ -217,8 +217,8 @@ void OnNecrachBloodKissRecieved()
-            var title = GameTexts.FindText("career_switch_title", careerObject.StringId);
-            var explaination = GameTexts.FindText("career_switch_explaination", careerObject.StringId);
+            var title = GameTexts.FindText("tor_career_switch_title", careerObject.StringId);
+            var explaination = GameTexts.FindText("tor_career_switch_explaination", careerObject.StringId);
```
explanation, no 'i' after the first 'a' despite it being present in the root word

SlyDevil commented on CSharpSourceCode/CampaignMechanics/CustomResourceBehavior/EonirFavorEnvoyTownBehavior.cs:
```diff
@@ -122,9 +122,9 @@ private void AddSpellsingerEnvoyDialogLines(CampaignGameStarter campaignGameStar
+        campaignGameStarter.AddPlayerLine("spellsinger_envoy_main_hub_spellsinger_magic", "spellsinger_envoy_main_hub", "back_to_main_hub_spellsinger", "{=str_tor_spelltrainer_eonir_open_book}I seek to further my knowledge of the Winds of Magic, can you help me achieve this?", () => MobileParty.MainParty.HasSpellCasterMember() && Hero.MainHero.Culture.StringId == TORConstants.Cultures.EONIR && Is ...
 
-        campaignGameStarter.AddPlayerLine("spellsinger_envoy_main_hub_spellsinger_lores", "spellsinger_envoy_main_hub", "spellsinger_envoy_spellsinger_lores", "{=tor_spelltrainer_eonir_open_book_str}I think i am ready to learn a new facet of the winds of magic.", () => IsSpellsingerEnvoy() && GreylordIsNotAllowedToLearnMoreLores(), null, 200, null);
+        campaignGameStarter.AddPlayerLine("spellsinger_envoy_main_hub_spellsinger_lores", "spellsinger_envoy_main_hub", "spellsinger_envoy_spellsinger_lores", "{=str_tor_spelltrainer_eonir_open_book}I think i am ready to learn a new facet of the winds of magic.", () => IsSpellsingerEnvoy() && GreylordIsNotAllowedToLearnMoreLores(), null, 200, null);
```
identical localization id to previous player line despite one being about spells and the other about lores

also, is "GreylordIsNotAllowedToLearnMoreLores" correctly named? If that returns true, then the player line is available to learn more lores despite the name implying the opposite.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -217,49 +217,49 @@ private void SpellsingerDialogs(CampaignGameStarter obj)
+            obj.AddPlayerLine("trainer_spellsinger_learnmagic_second_lore", "choices_spellsinger", "trainer_spellsinger_learnmagic_answer_secondlore", "{=str_tor_spelltrainer_spellsinger_learn_magic_warden_second_lore} I want to enhance my magic capabilities.", () => Hero.MainHero.HasCareer(TORCareers.Warden) &&
                 Hero.MainHero.HasUnlockedCareerChoiceTier(3) && Hero.MainHero.GetSkillValue(TORSkills.Spellcraft) > 200, null, 200, null);
-            obj.AddDialogLine("trainer_spellsinger_learnmagic_answer_secondlore", "trainer_spellsinger_learnmagic_answer_secondlore", "trainer_spellsinger_learnmagic_secondlore__answer_player", "{=tor_spelltrianer_spellsinger_learn_magic_warden_second_lore_answer_str}You truely choose the path of ariel. But the forest demand another tribute. Pay it and I will teach you more", isSpellsingerTrainer ...
+            obj.AddDialogLine("trainer_spellsinger_learnmagic_answer_secondlore", "trainer_spellsinger_learnmagic_answer_secondlore", "trainer_spellsinger_learnmagic_secondlore__answer_player", "{=str_tor_spelltrainer_spellsinger_learn_magic_warden_second_lore_answer}You truely choose the path of ariel. But the forest demand another tribute. Pay it and I will teach you more", isSpellsingerTrainer ...
```
Capital 'A' ariel.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -217,49 +217,49 @@ private void SpellsingerDialogs(CampaignGameStarter obj)
                 Hero.MainHero.HasUnlockedCareerChoiceTier(3), null, 200, null);
 
-            obj.AddPlayerLine("trainer_spellsinger_learnlore", "choices_spellsinger", "choices_spellsinger", "{=tor_spelltrainer_spellsinger_open_book_str}Teach me one of Ariels many pathways.", () => MobileParty.MainParty.HasSpellCasterMember() && spellsingerCondition() && SpellsingerAdditonalLoreCondition(), AdditionalLoresPrompt, 200, null);
+            obj.AddPlayerLine("trainer_spellsinger_learnlore", "choices_spellsinger", "choices_spellsinger", "{=str_tor_spelltrainer_spellsinger_open_book}Teach me one of Ariels many pathways.", () => MobileParty.MainParty.HasSpellCasterMember() && spellsingerCondition() && SpellsingerAdditonalLoreCondition(), AdditionalLoresPrompt, 200, null);
```
Ariel's - possessive apostrophe

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -217,49 +217,49 @@ private void SpellsingerDialogs(CampaignGameStarter obj)
 
-            obj.AddDialogLine("spellweaver_companion_choice_dialog", "spellweaver_companion_choice_dialog", "spellweaver_choice_lores_companion", "{=tor_spelltrainer_spellsinger_goodbye_str}A Highweaver, or Darkweaver... tough choice", isSpellsingerTrainer, null, 200, null);
-            obj.AddPlayerLine("spellweaver_choice_lores_companion", "spellweaver_choice_lores_companion", "choices_spellsinger", "{=tor_spelltrainer_spellsinger_open_book_str}Let me choose.", () => MobileParty.MainParty.HasSpellCasterMember() && spellsingerCondition(), spellweaverCompanionPrompt, 200, null);
+            obj.AddDialogLine("spellweaver_companion_choice_dialog", "spellweaver_companion_choice_dialog", "spellweaver_choice_lores_companion", "{=str_tor_spelltrainer_spellsinger_goodbye}A Highweaver, or Darkweaver... tough choice", isSpellsingerTrainer, null, 200, null);
```
This isn't a goodbye.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -217,49 +217,49 @@ private void SpellsingerDialogs(CampaignGameStarter obj)
 
-            obj.AddDialogLine("trainer_spellsinger_weaver", "spellweaver_choice_dialog", "spellweaver_choice_player", "{=tor_spelltrainer_spellsinger_goodbye_str}A spellsinger, can pick either the pathway of the Darkweaver or the one of the Highweaver. Choose wisely", isSpellsingerTrainer, null, 200, null);
-            obj.AddPlayerLine("spellweaver_choice_player", "spellweaver_choice_player", "choices_spellsinger", "{=tor_spelltrianer_spellweaver_choice_str}Let me choose.", () => MobileParty.MainParty.HasSpellCasterMember() && spellsingerCondition(), spellweaverPrompt, 200, null);
+            obj.AddDialogLine("trainer_spellsinger_weaver", "spellweaver_choice_dialog", "spellweaver_choice_player", "{=str_tor_spelltrainer_spellsinger_goodbye}A spellsinger, can pick either the pathway of the Darkweaver or the one of the Highweaver. Choose wisely", isSpellsingerTrainer, null, 200, null);
```
Not goodbye.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -217,49 +217,49 @@ private void SpellsingerDialogs(CampaignGameStarter obj)
-            obj.AddDialogLine("trainer_spellsinger_afterlearnspells", "openbook_spellsinger", "hub_spellsinger", "{=tor_spelltrainer_spellsinger_close_book_str}A new Facette of Ariels infinte knowledge.", null, openbookconsequence, 200, null);
+            obj.AddPlayerLine("trainer_spellsinger_playergoodbye", "choices_spellsinger", "saygoodbye", "{=str_tor_spelltrainer_spellsinger_player_goodbye}Ariel with you. ", null, null, 200, null);
+            obj.AddDialogLine("trainer_spellsinger_goodbye", "saygoodbye", "close_window", "{=str_tor_spelltrainer_spellsinger_goodbye}May Ariel guide you on all your paths through her garden.", isSpellsingerTrainer, null, 200, null);
+            obj.AddDialogLine("trainer_spellsinger_afterlearnspells", "openbook_spellsinger", "hub_spellsinger", "{=str_tor_spelltrainer_spellsinger_close_book}A new Facette of Ariels infinte knowledge.", null, openbookconsequence, 200, null);
```
Ariel's - possessive

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -350,7 +350,7 @@ void spellweaverCompanionPrompt()
                 }
 
-                var inquirydata = new MultiSelectionInquiryData(new TextObject("{=tor_magic_lore_prompt_label_str}Choose Companion").ToString(), new TextObject("{=tor_magic_lore_prompt_description_str}Which companion should become a Spellcaster?.").ToString(), list, true, 1, 1, "Confirm", "Cancel", OnChooseCompanion, null);
+                var inquirydata = new MultiSelectionInquiryData(new TextObject("{=str_tor_magic_lore_prompt_label}Choose Companion").ToString(), new TextObject("{=str_tor_magic_lore_prompt_description}Which companion should become a Spellcaster?.").ToString(), list, true, 1, 1, "Confirm", "Cancel", OnChooseCompanion, null);
```
shares localization id with line 318 but lore vs companion

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -370,7 +370,7 @@ void spellweaverCompanionChooseLore(Hero hero)
                 }
 
-                var inquirydata = new MultiSelectionInquiryData(new TextObject("{=tor_magic_lore_prompt_label_str}Choose Lore for your companion").ToString(), new TextObject("{=tor_magic_lore_prompt_description_str}Choose a lore to specialize in.").ToString(), list, true, 1, 1, "Confirm", "Cancel", args => OnChooseCompanionLore(hero, args), OnCancelLore);
+                var inquirydata = new MultiSelectionInquiryData(new TextObject("{=str_tor_magic_lore_prompt_label}Choose Lore for your companion").ToString(), new TextObject("{=str_tor_magic_lore_prompt_description}Choose a lore to specialize in.").ToString(), list, true, 1, 1, "Confirm", "Cancel", args => OnChooseCompanionLore(hero, args), OnCancelLore);
```
shares localization id with above, but text is different all 3 times

description localization id is different for 1 of 3

SlyDevil commented on CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:
```diff
@@ -454,29 +454,29 @@ private void AddDialogs(CampaignGameStarter obj)
-                "{=tor_spelltrainer_magictest_vc_vampire_player_specialize_lore_str}Even if you have the everliving-gift, the access to forbidden knowledge is restricted by my master. Provide us a gift and I am not speaking of gold, " +
+                "{=str_tor_spelltrainer_magictest_vc_vampire_player_specialize_lore}Even if you have the everliving-gift, the access to forbidden knowledge is restricted by my master. Provide us a gift and I am not speaking of gold, " +
                 "I need you to pay with the essence of Life. Then my masters can give you access... (1500{DARKENERGYICON}) ", null, null, 200, null);
             obj.AddPlayerLine("agree_dark_energy_price", "accept_dark_energy_price", "specializelore",
```
price differs from the line just before it

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/CursedSiteMenuLogic.cs:
```diff
@@ -165,14 +165,14 @@ private bool GhostsCondition(MenuCallbackArgs args)
             if (freeSlots <= 0)
             {
-                args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_not_enough_free_slots_text_str}You have not enough space in your party.");
+                args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_not_enough_free_slots}You have not enough space in your party.");
```
You don't have enough...

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/CursedSiteMenuLogic.cs:
```diff
@@ -156,7 +156,7 @@ private bool GhostsCondition(MenuCallbackArgs args)
         if (!(Hero.MainHero.PartyBelongedTo.GetMemberHeroes().Any(x => x.IsNecromancer()) || Hero.MainHero.IsVampire()))
         {
-            args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_not_necromancer_text_str}You are not a practitioner of necromancy.");
+            args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_not_necromancer}You are not a practitioner of necromancy.");
```
"You need a practitioner of necromancy." perhaps?
If a necromancer is in the party, it counts, but the text suggests that it's only the player that is considered.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/CursedSiteMenuLogic.cs:
```diff
@@ -165,14 +165,14 @@ private bool GhostsCondition(MenuCallbackArgs args)
             if (lastGhostRecruitmentTime >= (int)CampaignTime.Now.ToDays)
             {
-                args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_once_a_day_text_str}You can only perform this action once a day.");
+                args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_once_a_day}You can only perform this action once a day.");
```
"str_tor_custom_settlement_cursed_site_ghost_cooldown" I find to be a better id because :
- what's once a day?
- will it always be once a day?
- is that string going to be something generic where we would use it in multiple contexts for cursed sites, or it can be specific and not require an id change/potential confusion if we change the cooldown? (or would we deprecate it and create a "twice_a_day" string to replace it)

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/CursedSiteMenuLogic.cs:
```diff
@@ -188,7 +188,7 @@ private bool EmpoweringUndeadCondition(MenuCallbackArgs args)
         if (!(Hero.MainHero.PartyBelongedTo.GetMemberHeroes().Any(x => x.IsNecromancer()) || Hero.MainHero.IsVampire()))
         {
-            args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_not_necromancer_text_str}You are not a practitioner of necromancy.");
+            args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_not_necromancer}You are not a practitioner of necromancy.");
```
same comment as line 168

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/OakOfAgesMenuLogic.cs:
```diff
@@ -256,7 +256,7 @@ private void AddTreeSpiritMenu(CampaignGameStarter starter)
             false);     //check if left side only contain tree spirits done button condition
 
-        starter.AddWaitGameMenu("oak_of_ages_tree_spirits_menu_bind_dryads", "{=tor_custom_settlement_cursed_site_ghosts_progress_str}Performing binding ritual...",
+        starter.AddWaitGameMenu("oak_of_ages_tree_spirits_menu_bind_dryads", "{=str_tor_custom_settlement_cursed_site_ghosts_progress}Performing binding ritual...",
```
that doesn't look like the right localization id, and if it is, it should instead be something like, "str_tor_custom_settlement_binding_in_progress" to remain generic for its multiple usages

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/OakOfAgesMenuLogic.cs:
```diff
@@ -672,7 +672,7 @@ private bool RootUpgradeCondition(MenuCallbackArgs args, int upgradeCost, string
             {
                 args.IsEnabled = false;
-                args.Tooltip = new TextObject("{=tor_custom_settlement_we_party_size_info_str}Already unlocked.");
+                args.Tooltip = new TextObject("{=str_tor_custom_settlement_we_not_enough_harmony}Already unlocked.");
```
str_tor_custom_settlement_previously_unlocked

doesn't even need "we" mention as that's generic across other potential settlements

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/OakOfAgesMenuLogic.cs:
```diff
@@ -707,7 +707,7 @@ private bool RootAccessibleCondition(MenuCallbackArgs args, int travelCost, stri
         }
 
-        args.Tooltip = new TextObject("{=tor_custom_settlement_we_party_size_info_str}Not enough Harmony for traveling.");
+        args.Tooltip = new TextObject("{=str_tor_custom_settlement_we_not_enough_harmony}Not enough Harmony for traveling.");
```
shares a localization id with line 688

"str_tor_custom_settlement_we_not_enough_harmony_travel" perhaps

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/ShrineMenuLogic.cs:
```diff
@@ -119,7 +119,7 @@ private bool DefileCondtion(MenuCallbackArgs args)
             {
                 GameTexts.SetVariable("DEFILE_COOLDOWN_DAYS", DefillingCooldownInDays.ToString());
-                args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_once_a_day_text_str}You can only perform this action every {DEFILE_COOLDOWN_DAYS} days.");
+                args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_once_a_day}You can only perform this action every {DEFILE_COOLDOWN_DAYS} days.");
```
str_tor_custom_settlement_cursed_site_defile_cooldown

same as before :
- is the cooldown always going to be once a day? (it's not, the CD is currently 5 days)
- does it need to be generic for such a specific action?

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/ShrineMenuLogic.cs:
```diff
@@ -136,7 +136,7 @@ private bool DefileCondtion(MenuCallbackArgs args)
                 {
                     GameTexts.SetVariable("DEFILE_COOLDOWN_DAYS", DefillingCooldownInDays.ToString());
-                    args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_once_a_day_text_str}You can only perform this action every {DEFILE_COOLDOWN_DAYS} days.");
+                    args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_once_a_day}You can only perform this action every {DEFILE_COOLDOWN_DAYS} days.");
```
same as line 122; this can reuse the same defile text/id as the number of days isn't in the id, and it's set locally even if the value used in each is different.

the id could instead be "str_tor_custom_settlement_defile_cooldown" and it would cover shrines and cursed sites without potential confusion

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/ShrineMenuLogic.cs:
```diff
@@ -168,7 +168,7 @@ private bool LootCondition(MenuCallbackArgs args)
             {
                 GameTexts.SetVariable("DEFILE_COOLDOWN_DAYS", DefillingCooldownInDays.ToString());
-                args.Tooltip = new TextObject("{=tor_custom_settlement_cursed_site_once_a_day_text_str}You can only perform this action every {DEFILE_COOLDOWN_DAYS} days.");
+                args.Tooltip = new TextObject("{=str_tor_custom_settlement_cursed_site_once_a_day}You can only perform this action every {DEFILE_COOLDOWN_DAYS} days.");
```
another shrine defile that could use the generic id

do greenskins defile a cursed site to shut it down, or do they ward it?

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/ShrineMenuLogic.cs:
```diff
@@ -220,7 +220,7 @@ private bool PrayCondition(MenuCallbackArgs args)
             var god = ReligionObject.All.FirstOrDefault(x => x.StringId == careerGod);
             MBTextManager.SetTextVariable("CAREERGOD_NAME", god.DeityName);
-            args.Tooltip = new TextObject("{=tor_custom_settlement_shrine_blessing_already_active_str}You devoted your live to {CAREERGOD_NAME}. You can't pray here.", null);
+            args.Tooltip = new TextObject("{=str_tor_custom_settlement_shrine_blessing_already_active}You devoted your live to {CAREERGOD_NAME}. You can't pray here.", null);
```
str_tor_custom_settlement_shrine_not_devoted_god
_wrong_god
_invalid_god

I think the first is most clear. Nothing about this is blessing related.

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/CustomSettlementMenus/ShrineMenuLogic.cs:
```diff
@@ -267,14 +267,14 @@ private bool DonationCondition(MenuCallbackArgs args)
             var god = ReligionObject.All.FirstOrDefault(x => x.StringId == careerGod);
             MBTextManager.SetTextVariable("CAREERGOD_NAME", god.DeityName);
-            args.Tooltip = new TextObject("{=tor_custom_settlement_shrine_blessing_already_active_str}You devoted your live to {CAREERGOD_NAME}. You can't donate here.", null);
+            args.Tooltip = new TextObject("{=str_tor_custom_settlement_shrine_blessing_already_active}You devoted your live to {CAREERGOD_NAME}. You can't donate here.", null);
```
line 223 comment

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/TORCustomSettlementCampaignBehavior.cs:
```diff
@@ -198,12 +198,12 @@ private void OnMissionEnded(IMission obj)
                 }
 
-                var inq = new MultiSelectionInquiryData("Victory!", new TextObject("{=tor_custom_settlement_chaos_portal_victory_str}You are Victorious! Claim your reward! Select one!").ToString(), list, false, 1, 1, "OK", null, OnRewardClaimed, null);
+                var inq = new MultiSelectionInquiryData("Victory!", new TextObject("{=str_tor_custom_settlement_chaos_portal_victory}You are Victorious! Claim your reward! Select one!").ToString(), list, false, 1, 1, "OK", null, OnRewardClaimed, null);
```
"str_tor_custom_settlement_raid_victory" perhaps as this will apply to all 3 raid settlement types

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/TORCustomSettlementCampaignBehavior.cs:
```diff
@@ -198,12 +198,12 @@ private void OnMissionEnded(IMission obj)
             else
             {
-                var inq = new InquiryData("Defeated!", new TextObject("{=tor_custom_settlement_chaos_portal_lose_str}The enemy proved more than a match for you. Better luck next time!").ToString(), true, false, "OK", null, null, null);
+                var inq = new InquiryData("Defeated!", new TextObject("{=str_tor_custom_settlement_chaos_portal_lose}The enemy proved more than a match for you. Better luck next time!").ToString(), true, false, "OK", null, null, null);
```
"str_tor_custom_settlement_raid_lose" for a similar reason

SlyDevil commented on CSharpSourceCode/CampaignMechanics/TORCustomSettlement/TORSettlementComponents.cs:
```diff
@@ -259,7 +259,7 @@ public override void SpawnNewParty(out MobileParty party, Settlement initialTarg
             Clan beastmenClan = Clan.FindFirst(x => x.StringId == "beastmen_clan_1");
             var find = TORCommon.FindSettlementsAroundPosition(Settlement.Position.ToVec2(), 60, x => !x.IsRaided && !x.IsUnderRaid && x.IsVillage).GetRandomElementInefficiently();
-            var raidingParty = RaidingPartyComponent.CreateRaidingParty("beastmen_clan_1_party_" + RaidingPartyCount + 1, Settlement, new TextObject("{=tor_beastmen_raiders_str}Beastmen Raiders").ToString(), template, beastmenClan, MBRandom.RandomInt(75, 99));
+            var raidingParty = RaidingPartyComponent.CreateRaidingParty("beastmen_clan_1_party_" + RaidingPartyCount + 1, Settlement, new TextObject("{=str_tor_beastmen_raiders}Beastmen Raiders").ToString(), template, beastmenClan, MBRandom.RandomInt(75, 99));
```
i'm surprised I don't see an edit for the druuchi raiding party as well

SlyDevil commented on CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerButton/MercenaryCareerButtonBehavior.cs:
```diff
@@ -84,36 +84,36 @@ public override bool ShouldButtonBeVisible(CharacterObject characterObject, bool
             if (characterObject.IsTreeSpirit() || characterObject.IsUndead())
             {
-                displayText = new TextObject("{=tor_career_button_mercenary_noTreeUndead_str}Only works for humans or elves");
+                displayText = new TextObject("{=str_tor_career_button_mercenary_no_tree_undead}Only works for humans or elves");
```
doesn't matter for current release unless greenskin/dawi get access to merc career in which case this line won't make sense when they can promote an orc to companion

SlyDevil commented on CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerChoiceGroupObject.cs:
```diff
@@ -25,7 +27,27 @@ public CareerChoiceGroupObject(string stringId) : base(stringId) { }
+                {
+                    if (!nameOverride.Contains(nameText))
+                    {
+                        TORCommon.Log(String.Format("Career choice group name has been modified in externalized strings(tor_strings). this should be adapted in code: \n CODE {0} \n XML {1}", nameText, nameOverride), LogLevel.Warn);
```
I like the logging

Z3rca replied:
its now logged for all , so has been moved a little, but does still more or less the same thing

SlyDevil commented on CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerChoiceGroupObject.cs:
```diff
@@ -25,7 +27,27 @@ public CareerChoiceGroupObject(string stringId) : base(stringId) { }
+                }
+            }
+
+            base.Initialize(nameText, new TextObject("Choice group for " + name));
```
The 2nd part of this won't be localized. Not sure atm what exactly the difference is between what's in nameText and what is being put into the new TextObject for "name", but the entire contents of the TextObject is strings with no id which renders it unabled to be localized.

new TextObject(GameText.FindText("tor_career_choice_group") + nameText) possibly, then stick "Choice group for " into the xml with that id

Z3rca replied:
this description will not be displayed

SlyDevil commented on CSharpSourceCode/CharacterDevelopment/CareerSystem/CareerChoiceObject.cs:
```diff
@@ -33,50 +34,58 @@ public void Initialize(CareerObject ownerCareer, string description, string belo
-                    {
-
+                    
+                    text = new TextObject(descriptionOverride.Value.ToString());
```
There is a method call that can be performed on TextObjects to copy the contents from one to another. Basically equivalent to taking the TO, converting to string, then sticking into a new TO, but less hot potato.

Z3rca committed:
new TorTextHelper approach for setting up dialog text

Z3rca committed:
tools for translation

SlyDevil commented on CSharpSourceCode/Extensions/UI/TORMapInfoVMExtension.cs:
```diff
@@ -77,9 +77,9 @@ private List<TooltipProperty> GetCultureResourceHintText()
-            string artilleryDeployable = new TextObject("{=tor_ui_winds_of_magic_recharge_rate_str}Maximum Deployable Artillery Pieces:").ToString();
+            string artilleryTitle = new TextObject("{=str_tor_ui_artillery_title}Artillery").ToString();
+            string artilleryInventory = new TextObject("{=str_tor_ui_artillery_amount}Current Artillery Pieces in Inventory:").ToString();
+            string artilleryDeployable = new TextObject("{=str_tor_ui_winds_of_magic_recharge_rate}Maximum Deployable Artillery Pieces:").ToString();
```
str_tor_ui_artillery_maximum

SlyDevil commented on CSharpSourceCode/Extensions/UI/TORMapInfoVMExtension.cs:
```diff
@@ -107,9 +107,9 @@ private List<TooltipProperty> GetWindsHintText()
-            var blessingTitle = new TextObject("{=tor_ui_winds_of_magic_title_str}Blessing: ").ToString();
-            var durationTitle = new TextObject("{=tor_ui_winds_of_magic_maximum_str}Duration:").ToString();
-            var effect = new TextObject("{=tor_ui_winds_of_magic_recharge_rate_str}Effect:").ToString();
+            var blessingTitle = new TextObject("{=str_tor_ui_winds_of_magic_title}Blessing: ").ToString();
```
str_tor_ui_blessing_title

SlyDevil commented on CSharpSourceCode/Extensions/UI/TORMapInfoVMExtension.cs:
```diff
@@ -107,9 +107,9 @@ private List<TooltipProperty> GetWindsHintText()
-            var durationTitle = new TextObject("{=tor_ui_winds_of_magic_maximum_str}Duration:").ToString();
-            var effect = new TextObject("{=tor_ui_winds_of_magic_recharge_rate_str}Effect:").ToString();
+            var blessingTitle = new TextObject("{=str_tor_ui_winds_of_magic_title}Blessing: ").ToString();
+            var durationTitle = new TextObject("{=str_tor_ui_winds_of_magic_maximum}Duration:").ToString();
```
str_tor_ui_blessing_duration

SlyDevil commented on CSharpSourceCode/Extensions/UI/TORMapInfoVMExtension.cs:
```diff
@@ -107,9 +107,9 @@ private List<TooltipProperty> GetWindsHintText()
-            var effect = new TextObject("{=tor_ui_winds_of_magic_recharge_rate_str}Effect:").ToString();
+            var blessingTitle = new TextObject("{=str_tor_ui_winds_of_magic_title}Blessing: ").ToString();
+            var durationTitle = new TextObject("{=str_tor_ui_winds_of_magic_maximum}Duration:").ToString();
+            var effect = new TextObject("{=str_tor_ui_winds_of_magic_recharge_rate}Effect:").ToString();
```
str_tor_ui_blessing_effect
or maybe
str_tor_ui_effect_colon

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
             var KeyElementBindings = Module.CurrentModule.GlobalTextManager.GetGameText("str_key_name");
             var spellcastingModeKey = context + "_" + (int)TorKeyMap.QuickCastSelectionMenu;
-            KeyElementBindings.AddVariationWithId(spellcastingModeKey, new TextObject("{=tor_spellcasting_mode_key_str}Quick Cast Selection Menu"), new List<GameTextManager.ChoiceTag>());
+            KeyElementBindings.AddVariationWithId(spellcastingModeKey, new TextObject("{=str_tor_spellcasting_mode_key}Quick Cast Selection Menu"), new List<GameTextManager.ChoiceTag>());
```
"{=str_tor_ability_menu_key}Ability Selection Menu" or "{=str_tor_ability_wheel_key}Ability Selection Wheel" depending on generic "menu" vs. current "wheel" form used, or "{=str_tor_ability_mode_key}Ability Selection Mode" maybe?

I find the "mode" formulation to be the most accurate as it's used both for selecting abilities and casting. Though it's not only for spellcasting so I prefer the more generic "ability".

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
 
 
-            var spellCastingModeDescription = new TextObject("{=tor_spellcasting_mode_description_str}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
+            var spellCastingModeDescription = new TextObject("{=str_tor_spellcasting_mode_description}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
```
"str_tor_ability_mode_description"

If you tap Q, does it open the wheel, or does it immediately go into the reticle? Ie. is this more ability mode, or more aiming mode?

"1) During battle, press Q to enter aiming mode with a reticle for targeting your abilities. Aiming mode is accompanied by a visual animation on your character.
- slightly shorter formulation and "reticle" fix
- swapping verb tense to be more active : "press Q" is an instruction/call to action vs. "pressing Q will do..." is a passive description"

Z3rca replied:
the text might be outdated, we probably should rewokr that. unforunately needs to be hardcoded

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
 
-            var spellCastingModeDescription = new TextObject("{=tor_spellcasting_mode_description_str}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
+            var spellCastingModeDescription = new TextObject("{=str_tor_spellcasting_mode_description}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
                                                      "2) Using the mouse wheel will switch between your spells (if you have learned multiple)" + "\n" +
```
2) Move your mouse, or scroll the mouse wheel, to switch between your abilities (if you have multiple).
- active
- I didn't even know scrolling the mouse did that, I've always moved it around in the relevant direction.

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
-            var spellCastingModeDescription = new TextObject("{=tor_spellcasting_mode_description_str}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
+            var spellCastingModeDescription = new TextObject("{=str_tor_spellcasting_mode_description}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
                                                      "2) Using the mouse wheel will switch between your spells (if you have learned multiple)" + "\n" +
                                                      "3) New UI elements will appear on the bottom left of your screen, with an image representing your selected spell and a cooldown times " + "\n" +
```
3) New UI elements will appear on the bottom left of your screen : an image representing your selected spell, its cooldown time, your current Winds of Magic, and your career ability's charge.
- little bit more descriptive of what's all there
- minor fixes
- could add "(mana)" after WoM if that's wanted to be explicit

Z3rca replied:
same above

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
+            var spellCastingModeDescription = new TextObject("{=str_tor_spellcasting_mode_description}1) During battle pressing Q puts you into aiming mode, a reticule will appear and you will be able to target your spells. Aiming mode is accompanied by a visual animation on your character." + "\n" +
                                                      "2) Using the mouse wheel will switch between your spells (if you have learned multiple)" + "\n" +
                                                      "3) New UI elements will appear on the bottom left of your screen, with an image representing your selected spell and a cooldown times " + "\n" +
                                                      "4) Spells are reliant on the user's Winds of Magic (mana), this is fairly self-explanatory. It is regained over time on the campaign map.");
```
4) Spells require available Winds of Magic to cast. It is regained over time on the campaign map.
- mainly removing the "self-explanatory" quip; it's either evident in which case it's unnecessary, or it's not self-explanatory in which case it's wrong.

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
             KeyDescriptionElement.AddVariationWithId(spellcastingModeKey, spellCastingModeDescription, new List<GameTextManager.ChoiceTag>());
             KeyDescriptionElement.AddVariationWithId(quickCast, new TaleWorlds.Localization.TextObject(
-                "{=tor_quick_cast_description_str}Cast spells or prayers without needing to switch to spell cast mode. Warning: There is no target indication and the Spells follow predefined casting order. For targeted casting, always use the Spellcaster mode"), new List<GameTextManager.ChoiceTag>());
+                "{=str_tor_quick_cast_description}Cast spells or prayers without needing to switch to spell cast mode. Warning: There is no target indication and the Spells follow predefined casting order. For targeted casting, always use the Spellcaster mode"), new List<GameTextManager.ChoiceTag>());
```
Cast the last selected ability immediately. Warning: There is no target indication; for target sensitive casting, use the aiming mode."

SlyDevil commented on CSharpSourceCode/GameManagers/TORKeyInputManager.cs:
```diff
@@ -33,25 +33,25 @@ private static void InitializeHotKeyManager()
+                "{=str_tor_quick_cast_description}Cast spells or prayers without needing to switch to spell cast mode. Warning: There is no target indication and the Spells follow predefined casting order. For targeted casting, always use the Spellcaster mode"), new List<GameTextManager.ChoiceTag>());
             KeyDescriptionElement.AddVariationWithId(specialMove, new TaleWorlds.Localization.TextObject(
-                "{=tor_career_ability_description_str}Starting Career ability, like the mistform for the Vampire Counts, or Knightly strike of the Grailknight"), new List<GameTextManager.ChoiceTag>());
+                "{=str_tor_career_ability_description}Starting Career ability, like the mistform for the Vampire Counts, or Knightly strike of the Grailknight"), new List<GameTextManager.ChoiceTag>());
```
- Mistform
- Knightly Strike of the Grail Knight
Capitalization for proper nouns; I think "Grail" is an adjective on knight, rather than a compound word.

Z3rca committed:
wip merge all

SlyDevil commented on CSharpSourceCode/Quests/HuntCultistsQuestCampaignBehavior.cs:
```diff
@@ -76,35 +76,35 @@ public HuntCultistsIssue(Hero issueOwner, Settlement targetSettlement) : base(is
                 get
                 {
-                    TextObject textObject = new TextObject("{=tor_quest_hunt_cultist_issue_explanation_str}I need you to travel to {TARGET_SETTLEMENT}. Investigate the local populace and root out any cultists. On successful completion, the order will pay you {REWARD}{GOLD_ICON}.", null);
+                    TextObject textObject = new TextObject("{=str_tor_quest_hunt_cultist_issue_explanation}I need you to travel to {TARGET_SETTLEMENT}. Investigate the local populace and root out any cultists. On successful completion, the order will pay you {REWARD}{GOLD_ICON}.", null);
```
Do these grant prestige?
Could be something worth adding in the future.

SlyDevil commented on CSharpSourceCode/Quests/HuntCultistsQuestCampaignBehavior.cs:
```diff
@@ -76,35 +76,35 @@ public HuntCultistsIssue(Hero issueOwner, Settlement targetSettlement) : base(is
                 get
                 {
-                    TextObject textObject = new TextObject("{=tor_quest_hunt_cultist_description_str}Travel to target settlement and root out any cultist who may be hiding there.", null);
+                    TextObject textObject = new TextObject("{=str_tor_quest_hunt_cultist_description}Travel to target settlement and root out any cultist who may be hiding there.", null);
```
Travel to {TARGET_SETTLEMENT} and root out any cultist who may be hiding there."

May as well use the set variable if it's going to be there.

SlyDevil commented on CSharpSourceCode/Quests/HuntCultistsQuestCampaignBehavior.cs:
```diff
@@ -170,7 +170,7 @@ public HuntCultistsQuest(string questId, Hero questGiver, CampaignTime duration,
             }
 
-            public override TextObject Title => new TextObject("{=tor_quest_hunt_cultist_quest_title_str}A cultist in our midst");
+            public override TextObject Title => new TextObject("{=str_tor_quest_hunt_cultist_quest_title}A cultist in our midst");
```
"{=str_tor_quest_hunt_cultist_title}A cultist in our midst" ?

Line 101 has the same text with almost the same id. Are these titles likely to differ, or they can be identical in both english and any translated language by unifying the id?

SlyDevil commented on CSharpSourceCode/Quests/HuntCultistsQuestCampaignBehavior.cs:
```diff
@@ -228,7 +228,7 @@ private void SettlementEntered(MobileParty party, Settlement settlement, Hero he
                     if (settlement.IsUnderRaid || settlement.IsRaided)
                     {
-                        InquiryData data = new InquiryData("Village Raided", "{=tor_quest_hunt_cultist_village_raided_info_str}The village is raided, no chance to find any cultists now. Come back when the village is repopulated.", true, false, "OK", null, () => InformationManager.HideInquiry(), null);
+                        InquiryData data = new InquiryData("Village Raided", "{=str_tor_quest_hunt_cultist_village_raided_info}The village is raided, no chance to find any cultists now. Come back when the village is repopulated.", true, false, "OK", null, () => InformationManager.HideInquiry(), null);
```
The village is empty, no chance to investigate any cultists now. Perhaps they'll return when things are back to normal."
- Implies the cultists likely fled for the moment, rather than them having died and a cultist infestation is guaranteed to occur in any sufficiently populated region.

Z3rca committed:
modifying translation tool to add reports, allow to translate any file mentioned in the language xml

Z3rca committed:
move settlement editor to tools

Z3rca committed:
dialogs with new text system enchantment dialogs

also added skip validation option

Z3rca committed:
fixes

Z3rca committed:
fix oathgold and favor dialogs

Z3rca committed:
make use of TextHelper instead of having its own thing

Z3rca committed:
remove duplicate localization ids from name tags

the names were the same, there was no need for 2 pairs of ids.

Z3rca committed:
more moves to new translation system

Z3rca committed:
add obsolete check

Z3rca committed:
set TextHelper

Z3rca committed:
wip more scripts missing new texthelper method

Z3rca committed:
fix missing tags

Z3rca committed:
career button display

Z3rca committed:
more elements

Z3rca committed:
added rest of UI text with Text helper

Z3rca committed:
removed unused ids

Z3rca committed:
some additional informations

Z3rca committed:
git ignore

Z3rca committed:
fix exclusion of name duplicates
fix many report files being created
added to git ignore

Z3rca committed:
fix uses of old str pattern

Z3rca committed:
fixing string pattern of engineer quest.

Z3rca committed:
added missing entries

Z3rca committed:
Merge branch 'development' into ids

Z3rca committed:
not sure why this was in

Z3rca committed:
fixing slys issues

Z3rca committed:
some minor display text

Z3rca committed:
fixing cultist quest

Z3rca committed:
fixing remarked grammar issues add variable to text (issues from git)

Z3rca committed:
some issues

Z3rca committed:
added some missing strings

Z3rca committed:
Merge remote-tracking branch 'origin/development' into ids

Z3rca committed:
test

Z3rca committed:
fix

Z3rca committed:
merge fixes

Z3rca committed:
fixing more display texts

Z3rca committed:
missing namespaces

Z3rca committed:
had to hardcode, since not available at start

Z3rca committed:
added set former jibberish tags

Z3rca committed:
fixing wrong ids

Z3rca committed:
fix mainmenu patch
added some missing name spaces

Z3rca committed:
added more entries

Z3rca committed:
reworking scripts with game texts

added TextHelper pattern
use variable approach in contrast to hard coded strings

removed some unused ids from tor_strings

Z3rca committed:
use TextHelpers for companion dialogs

Z3rca committed:
added found missing entries

Z3rca committed:
working on fixing code strings to match xml

Z3rca committed:
fixes some texts in xml and code

Z3rca committed:
fix more text issues

Z3rca committed:
more fixes and missing strings
